### ISSUE TITLE
problems with paths or file names containing spaces

### ISSUE THREAD

**Issue by flokli** [invoice-x/invoice2data#133]
trying to read a file called "2018-05-09 foo.pdf", invoked via
```
invoice2data --input-reader tesseract 2018-05-09\ foo.pdf --output-format json --output-name foo --debug
```

```
convert: unable to open image `2018-05-09': No such file or directory @ error/blob.c/OpenBlob/2735.
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/504.
convert: unable to open image `Call': No such file or directory @ error/blob.c/OpenBlob/2735.
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/504.
convert: unable to open image `a': No such file or directory @ error/blob.c/OpenBlob/2735.
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/504.
convert: unable to open image `Bike': No such file or directory @ error/blob.c/OpenBlob/2735.
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/504.
convert: unable to open image `Rechnung2018-05-09.pdf': No such file or directory @ error/blob.c/OpenBlob/2735.
convert: no images defined `tiff:-' @ error/convert.c/ConvertImageCommand/3258.
Error in pixReadMem: size < 12
Error during processing.
```

**Comment by m3nu** [comment]
Thanks for the report. @duskybomb will look into it. Maybe due to the space that's not passed to Tesseract correctly.

**Comment by duskybomb** [comment]
I can confirm that this error is because of space and we cannot do anything about it, it's the problem with `imagemagick` module. Works fine with `pdftotext`
So just remove white-space from file name and you will be good to go @flokli

**Comment by m3nu** [comment]
Not true, @duskybomb.  While I avoid spaces in filenames, there is generally a way to escape spaces when passing them to the command line.

The issue is with these 2 lines:

```
    convert = "convert -density 350 %s -depth 8 tiff:-" % (path)
    p1 = subprocess.Popen(convert.split(' '), stdout=subprocess.PIPE)
```

This is terrible code because the command is built as string. The correct way is to build it as list and pass it to `subprocess`. E.g. ([from](https://docs.python.org/3/library/subprocess.html) the official docs)

`subprocess.run(["ls", "-l"])`

**Comment by duskybomb** [comment]
Will be using `Popen` instead of `run`, since run is not supported in python2. I tried this:
```
    convert = ['convert', '-density', '350', path, '-depth', '8', 'tiff:-']
    p1 = subprocess.Popen(convert.split(' '), stdout=subprocess.PIPE)
    
    tess = ['tesseract', 'stdin', 'stdout']
    p2 = subprocess.Popen(tess, stdin=p1.stdout, stdout=subprocess.PIPE)
```
and it works well. Should I continue with a PR?

**Comment by m3nu** [comment]
Yes please.

**Comment by m3nu** [comment]
`convert.split(' ')`

Not sure what this does or why it even works. Does a list have a `split()` function? Thought that's for strings?

**Comment by duskybomb** [comment]
I have removed `split()` don't worry

**Comment by duskybomb** [comment]
I think you need to fix CI for commits. It is not working?

**Comment by m3nu** [comment]
True. Reconnected it. Hope it works now. https://circleci.com/gh/invoice-x/invoice2data

**Comment by duskybomb** [comment]
Build failed on this `$ git push origin master --tags`

**Comment by flokli** [comment]
@m3nu any way to properly integrate this with github, so we see that next to PRs?

**Comment by m3nu** [comment]
It's already integrated. We recently moved the repo to a dedicated organization and the key wasn't set up yet. We used it for a long time before already.

See here: https://circleci.com/gh/invoice-x/invoice2data

**Comment by m3nu** [comment]
Fixed by @duskybomb via 332b111eb2e684249dc58fbb2f19e8deb437cdaa
